recipes/install.rb

```diff
@@ -28,6 +18,25 @@
   EOF
 end
 
+# When we upgrade to version 3.1.0 make sure we have set the subjects
+# of CAs
+if conda_helpers.is_upgrade && Gem::Version.new(node['install']['current_version']) < Gem::Version.new('3.1.0')
+  if node['hopsworks']['pki']['root']['name'].empty? || node['hopsworks']['pki']['intermediate']['name'].empty?
+    raise "It is an upgrade and Hopsworks CA subject name are not set. You have to set it to what was the previous names"
```

Comment:
Would it be better if we silently override the names with
root: "C=SE, ST=Sweden, L=Stockholm, O=SICS, CN=HopsRootCA"
intermediate: "C=SE, ST=Sweden, O=SICS, CN=HopsIntermediateCA"
kubernetes: "C=SE, ST=Sweden, L=Stockholm, O=LogicalClocks, CN=KubeHopsIntermediateCA"

They can't be different for versions < 3.1 and we won't have to set them for upgrades

Reply:
I don't think it's possible. Seems from the code that the configuration is then passed to the DML files. Which means that in all future updates we need to set those configurations, even if the current version is more than 3.1.0